File: src/core/input.rs
const KEY_COUNT: usize = 16;

pub struct Input {
    pub keys: [bool; KEY_COUNT]    
}

impl Input {
    pub fn new() -> Self {
        Self {
            keys: [false; KEY_COUNT]
        }
    }

    pub fn update(&mut self, key: usize, value: bool) {
        self.keys[key] = value;
    }
}

File: RetroC8/Sources/RetroC8/Input.swift
//
//  File.swift
//  
//
//  Created by Glenn Hevey on 20/11/2023.
//

import Foundation

struct Input {
    let keyCount: UInt = 16
    
    var keys: [Bool]
    
    init() {
        keys = [Bool](repeating: false, count: Int(keyCount))
    }
    
    mutating func update(key: UInt, value: Bool) {
        keys[Int(key)] = value
    }
}
